src/box/box.cc

```diff
@@ -3615,6 +3615,38 @@ on_wal_checkpoint_threshold(void)
 	gc_trigger_checkpoint();
 }
 
+static const uint32_t timeout_check_period = 1000;
+static uint32_t period = 0;
+
+/* Check periodically to see if the deadline of main cord has expired. */
+bool
+box_check_deadline()
+{
+	period = (period + 1) % timeout_check_period;
+	if (period == 0)
+		return check_deadline();
+	else
+		return true;
+}
+
+/* Force deadline of main cord on SIGURG. */
```

Comment:
AFAICS this signal is actually used by Linux to notify the process about urgent data on socket. You shouldn't reuse it.